pkg/ecdsa/tss/tss.go

```diff
@@ -169,3 +169,139 @@ func (s *ThresholdSigner) CalculateSignature(
+	handleLiquidationRecoveryMessage := func(netMsg net.Message) {
+		switch msg := netMsg.Payload().(type) {
+		case *LiquidationRecoveryMessage:
+			msgInChan <- msg
+		}
+	}
+	broadcastChannel.Recv(ctx, handleLiquidationRecoveryMessage)
+
+	memberBTCAddresses := make(map[string]string)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case msg := <-msgInChan:
+				for _, memberID := range group.groupMemberIDs {
+					if msg.SenderID.Equal(memberID) {
+						memberAddress, err := memberIDToAddress(
+							memberID,
+							pubKeyToAddressFn,
+						)
+						if err != nil {
+							logger.Errorf(
+								"could not convert member ID to address for "+
+									"a member of keep [%s]: [%v]",
+								group.groupID,
+								err,
+							)
+							break
+						}
+						memberBTCAddresses[memberAddress] = msg.BtcRecoveryAddress
+
+						logger.Infof(
+							"member [%s] from keep [%s] announced supplied btc address [%s] for liquidation recovery",
+							memberAddress,
+							group.groupID,
+							msg.BtcRecoveryAddress,
+						)
+
+						break
+					}
+				}
+
+				if len(memberBTCAddresses) == len(group.groupMemberIDs) {
+					cancel()
+				}
+			}
+		}
+	}()
+
+	go func() {
+		sendMessage := func() {
+			if err := broadcastChannel.Send(ctx,
+				&LiquidationRecoveryMessage{SenderID: group.memberID, BtcRecoveryAddress: "abc123"},
+			); err != nil {
+				logger.Errorf("failed to send btc recovery address: [%v]", err)
+			}
+		}
+
+		// Send the message first time. It will be periodically retransmitted
+		// by the broadcast channel for the entire lifetime of the context.
+		sendMessage()
+
+		<-ctx.Done()
+		// Send the message once again as the member received messages
+		// from all peer members but not all peer members could receive
+		// the message from the member as some peer member could join
+		// the protocol after the member sent the last message.
+		sendMessage()
+		return
+	}()
+
+	<-ctx.Done()
+
+	switch ctx.Err() {
+	case context.DeadlineExceeded:
+		for _, memberID := range group.groupMemberIDs {
+			memberAddress, err := memberIDToAddress(memberID, pubKeyToAddressFn)
+			if err != nil {
+				logger.Errorf(
+					"could not convert member ID to address for a member of "+
+						"keep [%s]: [%v]",
+					group.groupID,
+					err,
+				)
+				continue
+			}
+			if _, present := memberBTCAddresses[memberAddress]; !present {
+				logger.Errorf(
+					"member [%s] has not supplied a btc recovery address for keep [%s]; "+
+						"check if keep client for that operator is active and "+
+						"connected",
+					memberAddress,
+					group.groupID,
+				)
+			}
+		}
+		return fmt.Errorf(
+			"waiting for btc recovery addresses timed out after: [%v]", protocolReadyTimeout,
+		)
+	case context.Canceled:
+		logger.Infof("successfully gathered all btc addresses")
+
+		return nil
+	default:
+		return fmt.Errorf("unexpected context error: [%v]", ctx.Err())
+	}
+	return nil
```

Comment:
I think this line is unreachable and we can safely remove it.

Reply:
Feels like a good sign that `go vet` isn't running here.

@beaushinkle depending on how you went through setup here, it's possible you may not have pre-commit hooks set up (not sure if local-setup handles this step). If so, you may want to make sure to run `pre-commit install` in the repo root for keep-ecdsa. `brew install pre-commit` if it isn't already.